security/backends/elasticsearch/writer.py

```diff
@@ -55,9 +55,19 @@ def __len__(self):
 
 
 def get_querysets_by_batch(qs, batch):
-    steps = math.ceil(qs.count() / batch)
-    for step in range(steps):
-        yield qs[step * batch:(step+1) * batch]
+    last_document = None
+    while True:
+        batch_qs = qs
+        if last_document:
+            batch_qs = batch_qs.extra(search_after=last_document.meta.sort)
+
+        batch_list = list(batch_qs[:batch])
+        yield batch_list
```

Comment:
this works with similar way as cursor based pagination. Because over 10000 records base/offset pagination not working for elasticsearch